### PR TITLE
USB OTG driver for STM32L4

### PR DESCRIPTION
**Pull request by manuelbl**
So far libOpenCM3 assumes that all STM32L4 series chips have a USB FS device peripheral. However, this only holds for L41xx through L46xx while L47xx through L4Axx have an USB OTG peripheral.

This PR adds the necessary code so both the `otgfs_usb_driver` and `st_usbfs_v2_usb_driver` USB driver can be used.

The PR has been successfully tested with a NUCLEO-L476RG board.

To work without wiring the USB sensing pins), PR #1256 needs to be applied as well. In addition, 26ab78a needs to reverted (as seems to be the case for OTG peripherals with core ID 0x2000).

### CODE REVIEW

**Comment by manuelbl** [comment]
@karlp Can you help me with the build problems? Why does it fail with the *No rule to make target '../../../include/libopencm3/usb/hid_usage_tables.h', needed by 'usb_hid.o'* even though their is no reference to `hid_usage_tables.h` nor a file of that name?

**Comment by karlp** [comment]
don't stress the jenkins build failure, I'm having problems with that as well, I'm not sure of the cause right now.  whenever I try annd build manually on that machine it works normally. *shrugs*

File: include/libopencm3/stm32/l4/memorymap.h [hunk 1]
```diff
@@ -106,6 +106,7 @@
 #define GPIO_PORT_H_BASE		(PERIPH_BASE_AHB2 + 0x1c00)
 /* Still AHB2, good job ST */
 #define OTG_FS_BASE			(0x50000000U + 0x00000)
+#define USB_OTG_FS_BASE     OTG_FS_BASE
```
Comment (karlp) [inline include/libopencm3/stm32/l4/memorymap.h:109]: as OTG was never enabled on l4, just drop the original defintion, and use the "standard" definitoin please.

Comment (manuelbl) [reply include/libopencm3/stm32/l4/memorymap.h]: Ok, done

**Comment by manuelbl** [comment]
Does the Jenkins server have any special features (such as attached hardware) that couldn't be replaced by GitHub Actions? I would be willing to set it up.

**Comment by karlp** [comment]
yes, it has attached hardware, that's the entire point :)  it's purpose is to run the gadget0 usb tests on actual hardware.  It only has two boards at present, but more are planned.  the bizzare build errors are blocking that at present.

**Comment by karlp** [comment]
I'vee pulled this locally thanks.

**Comment by karlp** [comment]
This has landed in master now thanks!

**Pull request closed without merging**